feat: Require server and client to have the same NGO package version. [MTT-3048] (#2260)
MidLevel/MLAPI: com.unity.netcode.gameobjects/CHANGELOG.md

Additional documentation and release notes are available at [Multiplayer Documentation](https://docs-multiplayer.unity3d.com).

## [Unreleased]

### Changed

> - Client and server now need to have matching versions of Netcode for GameObjects. A client built with a different package version will now be rejected from connecting. (#2260)

Reviewer: ```suggestion
- Client and server must have fully matching versions of Netcode for GameObjects. A client built with a different package version will now be rejected from connecting. (#2260)
```

a bit more stronger language?
